pkg/app/pipectl/cmd/deployment/waitstatus.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployment
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pipe-cd/pipe/pkg/app/api/service/apiservice"
+	"github.com/pipe-cd/pipe/pkg/cli"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+type waitStatus struct {
+	root *command
+
+	deploymentID  string
+	status        []string
+	checkInterval time.Duration
+	timeout       time.Duration
+}
+
+func newWaitStatusCommand(root *command) *cobra.Command {
+	c := &waitStatus{
+		root:          root,
+		checkInterval: 15 * time.Second,
+		timeout:       15 * time.Minute,
+	}
+	cmd := &cobra.Command{
+		Use:   "wait-status",
+		Short: "Wait until the specified status.",
+		RunE:  cli.WithContext(c.run),
+	}
+
+	cmd.Flags().StringVar(&c.deploymentID, "deployment-id", c.deploymentID, "The deployment ID.")
+	cmd.Flags().StringSliceVar(&c.status, "status", c.status, "The list of waiting statuses. (PENDING|PLANNED|RUNNING|ROLLING_BACK|SUCCESS|FAILURE|CANCELLED)")
```

Comment:
nit: I thought it's better to automatically generate available statuses when building to reduce our maintenance cost. But it doesn't guarantee the order because of the map. So it's just for your reference, it's on you.
```go
var availableStatuses = func() string {
	statuses := ""
	for s := range model.DeploymentStatus_value {
		statuses = fmt.Sprintf("%s|%s", statuses, strings.TrimPrefix(s, "DEPLOYMENT_"))
	}
	return strings.TrimPrefix(statuses, "|")
}()
```